packages/components/ContextualMenu/src/SubmenuItem.tsx

```diff
@@ -0,0 +1,145 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { Image, View } from 'react-native';
+import {
+  SubmenuItemSlotProps,
+  SubmenuItemState,
+  SubmenuItemProps,
+  SubmenuItemRenderData,
+  submenuItemName,
+  SubmenuItemType,
+} from './SubmenuItem.types';
+import { compose, IUseComposeStyling } from '@uifabricshared/foundation-compose';
+import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
+import { Text } from '@fluentui-react-native/text';
+import { settings } from './SubmenuItem.settings';
+import { backgroundColorTokens, borderTokens, textTokens, foregroundColorTokens } from '@fluentui-react-native/tokens';
+import { mergeSettings } from '@uifabricshared/foundation-settings';
+import { useAsPressable, useKeyCallback, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { CMContext } from './ContextualMenu';
+
+export const SubmenuItem = compose<SubmenuItemType>({
+  displayName: submenuItemName,
+  usePrepareProps: (userProps: SubmenuItemProps, useStyling: IUseComposeStyling<SubmenuItemType>) => {
+    const {
+      disabled,
+      itemKey,
+      icon,
+      text,
+      accessibilityLabel = userProps.text,
+      onClick,
+      onHoverIn,
+      testID,
+      componentRef = React.useRef(null),
+      ...rest
+    } = userProps;
+
+    // Grabs the context information from Submenu (currently selected menuItem and client's onItemClick callback)
+    const context = React.useContext(CMContext);
+
+    const onItemClick = React.useCallback(
+      (e) => {
+        if (!disabled && (onClick || context.onItemClick)) {
+          context ?.onDismissMenu();
+          if (onClick) {
+            onClick();
+          } else {
+            context.onItemClick && context.onItemClick(itemKey);
+          }
+          e.stopPropagation();
+        }
+      },
+      [context, disabled, itemKey, onClick],
+    );
+
+    const cmRef = useViewCommandFocus(componentRef);
+    //const cmRef = React.useRef(null);
+    const onItemHoverIn = React.useCallback(
+      (e) => {
+        componentRef.current.focus();
+        onHoverIn(e);
```

Comment:
Is this an optional prop? Does onHoverIn() always exist?

Reply:
onHoverIn should always exist since SubmenuItem extends a pressable